src/App.tsx

```diff
@@ -1,60 +1,160 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+// import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+export function usersInTodos(): Todo[] {
+  return todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId),
+  }));
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(usersInTodos());
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasChangeError, setHasChangeError] = useState(false);
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const validation = event.target.value
+      .replace(/[^a-zа-я\s\d]/gi, '');
+
+    setTitle(validation);
+    setHasTitleError(false);
+  };
+
+  const changeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.currentTarget.value);
+    setHasChangeError(false);
+  };
+
+  const resetInputSFields = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const checkEmptyFileds = () => {
+    if (!userId || !title.trim()) {
+      if (!userId) {
+        setHasChangeError(true);
+      }
+
+      if (!title.trim()) {
+        setHasTitleError(true);
+      }
+    }
+
+    return userId && title.trim();
+  };
+
+  const findTheNextId = () => {
+    let theBiggestId = 0;
+
+    todos.forEach(todo => {
+      if (todo.id > theBiggestId) {
+        theBiggestId = todo.id;
+      }
+    });
+
+    return theBiggestId + 1;
```

Comment:
I guess we do too much operations here. Logically the last todo has the max id value. So we can just take it and add 1

Reply:
agree!